com.unity.ml-agents/Runtime/Sampler.cs

```diff
@@ -0,0 +1,77 @@
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using Unity.MLAgents;
+using Unity.MLAgents.Inference.Utils;
+using UnityEngine;
+
+namespace Unity.MLAgents
+{
+
+    /// <summary>
+    /// Takes a list of floats that encode a sampling distribution and returns the sampling function.
+    /// </summary>
+    internal sealed class SamplerFactory
+    {
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        internal SamplerFactory()
```

Comment:
@vincentpierre thoughts on removing this empty constructor completely?

Reply:
My view would be to have one Sampler of each type since we will only ever use one of the functions provided.

Reply:
Whoops, looks like github ate my comment. 

How about making the class (and methods) static instead?

Reply:
Done.